src/main/scala/com/yotpo/metorikku/metric/Metric.scala

```diff
@@ -93,8 +103,14 @@ case class Metric(configuration: Configuration, metricDir: File, metricName: Str
     val tags = Map("metric" -> metricName, "dataframe" -> dataFrameName, "output_type" -> outputType.toString)
     instrumentationProvider.count(name="counter", value=dataFrameCount, tags=tags)
     log.info(s"Starting to Write results of ${dataFrameName}")
+    val currentTime = System.currentTimeMillis
     try {
       writer.write(dataFrame)
+      val epochLagTimeValue = getEpochLagTime(dataFrame, lagReporterTimeColumn)
```

Comment:
Please wrap it in a function and flag it with a boolean, not with the name of the time column